### DISCUSSION TITLE
Support sync:false for useLocalStorage hook

### DISCUSSION THREAD

**Discussion by soorajshankar** [mantinedev/mantine#7217] [Feature requests]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.14.1

### What package has an issue?

@mantine/hooks

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

All

### Describe the bug

it would be nice to have a way to control the [sync](https://mantine.dev/hooks/use-local-storage/#browser-tabs-synchronization) behavior, right now the sync is enabled across tabs by default. 

It would be nice if it can be controlled in this way 

```
  const [val, setVal] = useLocalStorage<string>({
    defaultValue: '',
    sync: false,
    key
  });
```

### Possible fix

add another hook prop sync that defaults to true, and make sure it [dispatch storage event ](https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/hooks/src/use-local-storage/create-storage.ts#L114)only if the sync is true. 

### Self-service

- [X] I would be willing to implement a fix for this issue

**Comment by rtivital** [comment]
This feature will be available in 7.16 minor release on January 14.
